asserts/snap_asserts.go

```diff
@@ -698,3 +698,121 @@ func InitBuiltinBaseDeclaration(headers []byte) error {
 	builtinBaseDeclaration = a.(*BaseDeclaration)
 	return nil
 }
+
+// TODO(matt): consider renaming this ... Developer is a bit vague.
+type Developer struct {
+	AccountID string
+	Since     time.Time
+	Until     time.Time
+}
+
+// SnapDeveloper holds a snap-developer assertion, defining the developers who
+// can collaborate on a snap while it's owned by a specific publisher.
+type SnapDeveloper struct {
+	assertionBase
+	developers []*Developer
+}
+
+// SnapID returns the snap id of the snap.
+func (snapdev *SnapDeveloper) SnapID() string {
+	return snapdev.HeaderString("snap-id")
+}
+
+// PublisherID returns the publisher's account id.
+func (snapdev *SnapDeveloper) PublisherID() string {
+	return snapdev.HeaderString("publisher-id")
+}
+
+// Developers returns the developers allowed to collaborate on the snap.
+func (snapdev *SnapDeveloper) Developers() []*Developer {
+	return snapdev.developers
+}
+
+func (snapdev *SnapDeveloper) checkConsistency(db RODatabase, acck *AccountKey) error {
+	// Check authority is the publisher or trusted.
+	authorityID := snapdev.AuthorityID()
+	publisherID := snapdev.PublisherID()
+	if !db.IsTrustedAccount(authorityID) && (publisherID != authorityID) {
+		return fmt.Errorf("snap-developer must be signed by the publisher or a trusted authority: authority-id=%q, publisher-id=%q", authorityID, publisherID)
```

Comment:
This sentence would end better with something like

`... or a trusted authority, but got authority %q and publisher %q.`

at the moment it feels like it's telling what the trusted authority is, rather than reporting what it got instead.